static/js/apps/eval_retrieval_generation/sxs/data_store.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ Copyright 2024 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import { doc, DocumentReference, getDoc, setDoc } from "firebase/firestore";
+
+import { db } from "../../../utils/firebase";
+import { Rating, SxsPreference } from "./types";
+
+/**
+ * Gets the document ref for a particular rating.
+ */
+function getRatingRef(
+  sheetIdA: string,
+  sheetIdB: string,
+  queryId: number,
+  sessionId: string
+): DocumentReference {
+  // Use sheet IDs in lexicographical order so it doesn't matter which order
+  // they're passed in.
+  const firstSheetId = sheetIdA < sheetIdB ? sheetIdA : sheetIdB;
+  const secondSheetId = sheetIdA < sheetIdB ? sheetIdB : sheetIdA;
+  return doc(
+    db,
+    "sheets",
+    `${firstSheetId}:${secondSheetId}`,
+    "queries",
+    queryId.toString(),
+    "ratings",
+    String(sessionId)
+  );
+}
+
+/**
```

Comment:
wonder if these 2 functions are even necessary because we can just:

- update getRatingRef to getRatingRefPath
- use getAllFields and setFields from here: https://github.com/datacommonsorg/website/blob/master/static/js/apps/eval_retrieval_generation/data_store.tsx#L63-L77

Reply:
I forked them to avoid using `{merge: true}` and also because I worry it might be unexpected for SxS data store methods to rely on regular eval tool data store methods, since some regular store methods make assumptions about data being for regular eval tool (e.g. saveToStore assumes a regular eval path).

Reply:
ok makes sense!